features/steps/thens/relations.py

```diff
@@ -131,6 +126,157 @@ def step_impl(context, related, relating, other_entity, condition):
                         yield(err.RuleSuccessInst(True, inst))
 
 
+@then('The IfcPropertySet Name attribute value must use predefined values according to the {table} table')
+@then('The IfcPropertySet must be assigned according to the property set definitions table {table}')
+@then('Each associated IfcProperty must be named according to the property set definitions table {table}')
+@then('Each associated IfcProperty must be of type according to the property set definitions table {table}')
+@then('Each associated IfcProperty value must be of data type according to the property set definitions table {table}')
```

Comment:
I think we could use something more elegant to parse these sentences. But that would be for another PR 
https://github.com/buildingSMART/ifc-gherkin-rules/pull/48

Reply:
Agreed. I have multiple then's here, as it was requested to have the checks listed in the .feature files, so:

      Then The IfcPropertySet Name attribute value must use predefined values according to the IFC4X3_ADD2_definitions.csv table
      Then The IfcPropertySet must be assigned according to the property set definitions table IFC4X3_ADD2_definitions.csv
      Then Each associated IfcProperty must be named according to the property set definitions table IFC4X3_ADD2_definitions.csv
      Then Each associated IfcProperty must be of type according to the property set definitions table IFC4X3_ADD2_definitions.csv
      Then Each associated IfcProperty value must be of data type according to the property set definitions table IFC4X3_ADD2_definitions.csv
      
      Instead of 
      
      Then The IfcPropertySet must use predefined values according to the IFC4X3_ADD2_definitions.csv table
      
      
Which makes sense I think, as it's way more explicit and clear for the user what is checked. 
But then again I don't want to write multiple steps in .py, as it's so compact of a check, that it makes sense in my opinion to do it in one function. That's why I have multiple decorators.

Reply:
I will resolve the conversation, although I agree, that something better can be done here (but I don't know what without major code changes to the whole repository)